src/utils.js

```diff
@@ -826,38 +826,73 @@ export function timestamp() {
   return new Date().getTime();
 }
 
+/**
+ * When the deviceAccess flag config option is false, no cookies should be read or set
+ * @returns {boolean}
+ */
+export function hasDeviceAccess() {
+  return config.getConfig('deviceAccess') !== false;
+}
+
+/**
+ * @returns {(boolean|undefined)}
+ */
 export function checkCookieSupport() {
   if (window.navigator.cookieEnabled || !!document.cookie.length) {
     return true;
   }
 }
+
+/**
+ * @returns {boolean}
+ */
 export function cookiesAreEnabled() {
-  if (internal.checkCookieSupport()) {
-    return true;
+  if (hasDeviceAccess()) {
+    if (internal.checkCookieSupport()) {
+      return true;
+    }
+    window.document.cookie = 'prebid.cookieTest';
+    return window.document.cookie.indexOf('prebid.cookieTest') !== -1;
   }
-  window.document.cookie = 'prebid.cookieTest';
-  return window.document.cookie.indexOf('prebid.cookieTest') != -1;
+  return false;
 }
 
+/**
+ * @param {string} name
+ * @returns {(string|null)}
+ */
 export function getCookie(name) {
-  let m = window.document.cookie.match('(^|;)\\s*' + name + '\\s*=\\s*([^;]*)\\s*(;|$)');
-  return m ? decodeURIComponent(m[2]) : null;
+  if (hasDeviceAccess()) {
+    let m = window.document.cookie.match('(^|;)\\s*' + name + '\\s*=\\s*([^;]*)\\s*(;|$)');
+    return m ? decodeURIComponent(m[2]) : null;
+  }
+  return null;
 }
 
-export function setCookie(key, value, expires, sameSite) {
-  document.cookie = `${key}=${encodeURIComponent(value)}${(expires !== '') ? `; expires=${expires}` : ''}; path=/${sameSite ? `; SameSite=${sameSite}` : ''}`;
+/**
+ * @param {string} key
+ * @param {string} value
+ * @param {string} [expires='']
+ * @param {string} [sameSite='/']
+ * @param {(string|undefined)} [domain='']
+ */
+export function setCookie(key, value, expires, sameSite, domain) {
+  if (hasDeviceAccess()) {
+    document.cookie = `${key}=${encodeURIComponent(value)}${(expires !== '') ? `; expires=${expires}` : ''}; path=/${sameSite ? `; SameSite=${sameSite}` : ''}${domain ? `; domain=${domain}` : ''}`;
+  }
```

Comment:
Hi @idettman 

I had a quick question.

For this new `domain` property, are we expecting it to contain the full URL or just the actual domain?  

There's another PR where someone wants to use a `domain` argument to help populate the `path` property for the cookie we're writing.  They're doing the extra parsing the domain var to isolate the URI path (see [here](https://github.com/prebid/Prebid.js/pull/4803/files#diff-2b4ca49d4bb0a774c4d4c1672d7aa781R848) ).

I wanted to confirm how his variable should be populated to ensure that both eventual workflows match.

Thanks.

Reply:
**domain** is only used to set the cookie `;domain=domain` value
- (e.g., 'example.com' or 'subdomain.example.com').
- If not specified, defaults to the host portion of the current document location.
- If a domain is specified, subdomains are always included.
- domain must match the domain of the JavaScript origin. setting cookies to foreign domains will be silently ignored.
```

Reply:
I understand that this PR is planning to use the `domain` argument to populate the `domain` key in the cookie, but I was curious how we expect that argument to be populated?  

Are we going to expect it to be just the domain or the full URL?

Reply:
@jsnellbaker - it appears from the comment that the expectation is that it's just the domain.

>  * @param {string} [domain] domain (e.g., 'example.com' or 'subdomain.example.com').

And looking at the invibes code that prompted the feature, I don't see it dealing with the protocol.